### ISSUE TITLE
igxChip - improve focusable elements to allow the user to set context specific aria attributes.

### ISSUE THREAD

**Issue by MayaKirova** [IgniteUI/igniteui-angular#7248]
## Is your feature request related to a problem? Please describe.

Currently the main `igx-chip`'s focusable element is an internal div.
Hence if the user sets some  application context specific aria attributes on the main <igx-chip> element they will never be read by a screen reader since the root element is not focusable.

## Describe the solution you'd like

The igx-chip element itself should be focusable. 
TabIndex and keyboard interactions should be moved on the Host element from the internal div.

## Additional context
If the user wants to set some aria attributes like: title, aria-label, aria-describedby etc. he should be able to set them on the igx-chip and then have the screen reader read them.
